File: LeetCode/easy/Two_Sum.cxx
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::vector<int> indices(nums.size());
        std::iota(indices.begin(), indices.end(), 0);
        std::sort(indices.begin(), indices.end(), [&nums](int a, int b) {
            return nums[a] < nums[b];
        });
        std::sort(nums.begin(), nums.end());
        int l = 0, r = nums.size() - 1;
        while (1) // because it assumed that a solution exists
        {
            int s = nums[l] + nums[r];
            if (s == target)
                break;
            else if (s < target)
            {
                l++;
            }
            else
            {
                r--;                
            }
        }
        
        std::vector<int> res = {indices[l], indices[r]};
        return res;
    }
};

File: LeetCode/easy/Two_Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        sorted_nums = sorted(enumerate(nums), key=lambda x : x[1])

        l = 0
        r = len(nums) - 1
        while l < r:
            s = sorted_nums[l][1] + sorted_nums[r][1]
            if s == target:
                return sorted_nums[l][0], sorted_nums[r][0]
            elif s < target:
                l += 1
            else:
                r -= 1

